src/api/main/routing.ts

```diff
@@ -173,4 +173,110 @@ export type BestRouteResponse = {
   processingLimitReached: boolean
   walletNotSupportingFromBlockchain: boolean
+  confirmSwapStatus?: boolean
+}
+
+export type Tag =
+  | 'RECOMMENDED'
+  | 'FASTEST'
+  | 'LOWEST_FEE'
+  | 'HIGH_IMPACT'
+  | 'CENTRALIZED'
+
+/**
+ * A tag that can be assigned to a route.
+ */
+export type TagValue = Tag | Omit<string, Tag>
+
+/**
+ * The multi-routing result tag.
+ *
+ * @property {string} label - The human-readable label.
+ * @property {TagValue} value - The value of Tag.
+ */
+export type RouteTag = { label: string; value: TagValue }
+
+/**
+ * The type of preferences for sorting routes.
+ */
+export type PreferenceType = 'FEE' | 'SPEED' | 'PRICE' | 'NET_OUTPUT'
+
+/**
+ * Score calculated for each preference for sorting in UI
+ *
+ * @property {PreferenceType} preferenceType - Type of evaluated aspect.
+ * @property {number} score - Value of score (typically between 0 to 100).
+ */
+export type SimulationScore = {
+  preferenceType: PreferenceType
+  score: number
+}
+
+/**
+ * Full information of a path of multiple swaps that should be executed by user to swap X to Y
+ *
+ * @property {string} requestId - The unique requestId which is generated for this request by the server. It should be
+ * passed down to all other endpoints if this swap continues on. e.g. d10657ce-b13a-405c-825b-b47f8a5016ad
+ * @property {string} outputAmount - The estimation of Rango from output amount of Y
+ * @property {RoutingResultType} resultType
+ * @property {SwapResult[]} swaps - List of required swaps to swap X to Y with the expected outputAmount
+ * @property {SimulationScore[]} scores
+ * @property {RouteTag[]} tags
+ * @property {BlockchainValidationStatus[] | null} validationStatus - Pre-requisites check result. It will be null if
+ * the request checkPrerequisites was false
+ * @property {string[]} missingBlockchains - List of all blockchains which are necessary to be present for the best
+ * route and user has not provided any connected wallets for it. A null or empty list indicates that there is no problem.
+ * @property {boolean} walletNotSupportingFromBlockchain - A warning indicates that none of your wallets have the same
+ * blockchain as X asset
+ */
+export type MultiRouteSimulationResult = {
+  requestId: string
+  outputAmount: string
+  resultType: RoutingResultType
+  swaps: SwapResult[]
+  scores: { preferenceType: PreferenceType; score: number }[]
+  tags: RouteTag[]
+  validationStatus: BlockchainValidationStatus[] | null
+  missingBlockchains: string[] | null
+  walletNotSupportingFromBlockchain: boolean
+}
+
+/**
+ * The best route request body for multi-routing
+ */
+export type BestsRoutesRequest = BestRouteRequest
+
+/**
+ * The best route response for multi-routing, if the result fields is null, it means that no route is found
+ *
+ * @property {MultiRouteSimulationResult} results - List of best routes data
+ */
+export type BestsRouteResponse = BestRouteResponse & {
+  results: MultiRouteSimulationResult[]
+}
```

Comment:
I think we have some partial fields from `BestRouteResponse` not all of them in multi route response. (for example we don't have fields related to validation afaik) also you have missed some fields like `routeId`

Reply:
I included the routeId field in the MultiRouteResponse and verified the types. All remaining fields are identical between the BestRouteResponse and MultiRouteResponse.